fdb-record-layer-core/src/main/proto/record_cursor.proto

```diff
@@ -133,6 +133,21 @@ message RecursiveCursorContinuation {
     optional bytes activeStateContinuation = 3;
 }
 
+message AggregateCursorContinuation {
+    optional bytes continuation = 1;
+    optional PartialAggregationResult partial_aggregation_results = 2;
+}
+
+message PartialAggregationResult {
+  optional bytes group_key = 1;
+  repeated AccumulatorState accumulator_states = 2;
+}
+
+message AccumulatorState {
+  optional string physical_operator_name = 1;
+  repeated string state = 2; // 2 for avg
```

Comment:
I don't think `string` is the right data type for this. It seems like it would be better for the state object to be something like a `oneof` the following:

1. `int32`
2. `int64`
3. `float`
4. `double`
5. `bytes`

I think those are the value types currently supported by aggregates. I could also see `bytes` being used for `Message`s if we wanted to go down that route for average/record constructor values.

Reply:
Makes sense, modified.